components/internal/ApplicantStatusBadge.tsx

```diff
@@ -0,0 +1,118 @@
+import { Badge, Box } from '@chakra-ui/react'; // Chakra UI
+import Image from 'next/image'; // Optimized images
+
+type Props = {
+  variant: string;
```

Comment:
We can make this a union of strings:
`variant: "completed" | "expiring" | "rejected" | ...`
to enforce the allowed variants